src/sidebar/components/test/user-menu-test.js

```diff
@@ -0,0 +1,248 @@
+        const wrapper = createUserMenu();
+
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        assert.isFunction(profileMenuItem.prop('onClick'));
+      });
+    });
+
+    context('third-party user', () => {
+      beforeEach(() => {
+        fakeIsThirdPartyUser.returns(true);
+      });
+
+      it('should be disabled if no service configured', () => {
+        fakeServiceConfig.returns(null);
+
+        const wrapper = createUserMenu();
+
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        assert.isTrue(profileMenuItem.prop('isDisabled'));
+      });
+
+      it('should be disabled if service feature not supported', () => {
+        fakeServiceConfig.returns({ onProfileRequestProvided: false });
+
+        const wrapper = createUserMenu();
+
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        assert.isTrue(profileMenuItem.prop('isDisabled'));
+      });
+
+      it('should be enabled if service feature support', () => {
+        fakeServiceConfig.returns({ onProfileRequestProvided: true });
+
+        const wrapper = createUserMenu();
+
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        assert.isUndefined(profileMenuItem.prop('isDisabled'));
+      });
+
+      it('should have a callback if enabled', () => {
+        fakeServiceConfig.returns({ onProfileRequestProvided: true });
+
+        const wrapper = createUserMenu();
+
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        assert.isFunction(profileMenuItem.prop('onClick'));
+      });
+    });
+
+    describe('profile-selected callback', () => {
+      it('should fire profile event for third-party user', () => {
+        fakeServiceConfig.returns({ onProfileRequestProvided: true });
+        fakeIsThirdPartyUser.returns(true);
+        const wrapper = createUserMenu();
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        const onProfileSelected = profileMenuItem.prop('onClick');
+
+        onProfileSelected();
+
+        assert.equal(fakeBridge.call.callCount, 1);
+        assert.calledWith(fakeBridge.call, fakeProfileBridgeEvent);
+      });
+
+      it('should not fire profile event for first-party user', () => {
+        fakeIsThirdPartyUser.returns(false);
+        const wrapper = createUserMenu();
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        const onProfileSelected = profileMenuItem.prop('onClick');
+
+        onProfileSelected();
+
+        assert.equal(fakeBridge.call.callCount, 0);
+      });
+    });
+  });
+
+  describe('account settings menu item', () => {
+    it('should be present if first-party user', () => {
+      fakeIsThirdPartyUser.returns(false);
+
+      const wrapper = createUserMenu();
+
+      const accountMenuItem = findMenuItem(wrapper, 'Account settings');
+      assert.isTrue(accountMenuItem.exists());
```

Comment:
Ah, I didn't know you could call `exists()` on a wrapper with no args. That's useful to know :)